tests/integration/targets/ndo_ptp_policy/tasks/main.yml

```diff
@@ -0,0 +1,233 @@
+          - cm_update_ptp_policy.previous.global.adminState == "enabled"
+          - cm_update_ptp_policy.previous.global.fabSyncIntvl == -3
+          - cm_update_ptp_policy.current == cm_update_ptp_policy.proposed
+          - cm_update_ptp_policy.current.description == cm_update_ptp_policy.proposed.description == "changed_description"
+          - cm_update_ptp_policy.current.global.adminState == cm_update_ptp_policy.proposed.global.adminState == "disabled"
+          - cm_update_ptp_policy.current.global.fabSyncIntvl == cm_update_ptp_policy.proposed.global.fabSyncIntvl == -2
+          - cm_update_ptp_policy.current.global.prio1 == cm_update_ptp_policy.proposed.global.prio1 == 100
+          - cm_update_ptp_policy.current.global.prio2 == cm_update_ptp_policy.proposed.global.prio2 == 200
+          - cm_update_ptp_policy.current.global.globalDomain == cm_update_ptp_policy.proposed.global.globalDomain == 1
+          - nm_update_ptp_policy is changed
+          - nm_update_ptp_policy.current.description == "changed_description"
+          - nm_update_ptp_policy.current.global.adminState == "disabled"
+          - nm_update_ptp_policy.current.global.fabSyncIntvl == -2
+          - nm_update_ptp_policy.current.global.globalDomain == 1
+          - nm_update_ptp_policy.current.global.prio1 == 100
+          - nm_update_ptp_policy.current.global.prio2 == 200
+          - nm_update_ptp_policy_again is not changed
+
+    # QUERY
+    - name: Update PTP policy name with UUID
+      cisco.mso.ndo_ptp_policy:
+        <<: *update_ptp_policy
+        ptp_policy: ansible_test_ptp_policy_updated
+        ptp_policy_uuid: "{{ nm_add_ptp_policy.current.uuid }}"
+        state: present
+
+    - name: Query a PTP policy with ptp_policy name
+      cisco.mso.ndo_ptp_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        ptp_policy: ansible_test_ptp_policy_updated
+        state: query
+      register: query_one
+
+    - name: Query all PTP policies in the template
+      cisco.mso.ndo_ptp_policy:
+        <<: *mso_info
+        template: ansible_fabric_policy_template
+        state: query
+      register: query_all
+
+    - name: Verify query_one and query_all
+      assert:
+        that:
+          - query_all is not changed
+          - query_one is not changed
+          - query_all.current == query_one.current
+          - query_one.current.name == "ansible_test_ptp_policy_updated"
+          - query_one.current.description == "changed_description"
+          - query_one.current.global.adminState == "disabled"
+          - query_one.current.global.fabSyncIntvl == -2
+          - query_one.current.global.globalDomain == 1
+          - query_one.current.global.prio1 == 100
+          - query_one.current.global.prio2 == 200
+          - query_one.current.global.fabAnnounceTimeout == 3
+          - query_one.current.global.fabAnnounceIntvl == 1
+          - query_one.current.global.fabProfileTemplate == "aes67"
+
```

Comment:
Are there any ERROR test cases that can be added here? Potentially some testing the value ranges?

Reply:
The API returns an error message if the value passed is not in range and our code doesn't take of this, so no tests are necessary here